src/app/scenes/SceneStorage.h

```diff
@@ -0,0 +1,168 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <app/PersistentData.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/SceneId.h>
+#include <lib/support/Pool.h>
+
+namespace chip {
+namespace scenes {
+
+class SceneStorage
+{
+public:
+    static constexpr size_t kIteratorsMax = CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS;
+
+    struct FabricEndpoint
```

Comment:
Why do we have a separate structure that only contains an EndpointId? Why not use the endpointId type directly then? Or will this increase in size and content? Add documentation.

Same below.